pyccl/halos/halo_model.py

```diff
@@ -313,6 +328,8 @@ def I_0_2(self, cosmo, k, a, prof, *, prof2=None, prof_2pt):
         """
         if prof2 is None:
             prof2 = prof
+        self._fix_profile_mass_def(prof)
+        self._fix_profile_mass_def(prof2)
```

Comment:
How does this work? If `prof2` is not `None`, I think it will overwrite `prof`

Reply:
Same in all cases below

Reply:
`_fix_profile_mass_def` only defines the profile's `mass_def` if it is `None`. Otherwise it doesn't do anything. So the second call will not modify `prof`.

Reply:
hmm, and what if they don't have the same `mass_def`?

Reply:
An error will be thrown in the next line (334)